editor/src/messages/tool/tool_messages/select_tool.rs

```diff
@@ -115,137 +109,93 @@ impl LayoutHolder for SelectTool {
-					.widget_holder(),
-				IconButton::new("BooleanSubtractBack", 24)
-					.tooltip("Coming Soon: Boolean Subtract Back")
-					.disabled(deactivate_boolean_ops)
-					.on_update(|_| DialogMessage::RequestComingSoonDialog { issue: Some(1091) }.into())
-					.widget_holder(),
-				IconButton::new("BooleanIntersect", 24)
-					.tooltip("Coming Soon: Boolean Intersect")
-					.disabled(deactivate_boolean_ops)
-					.on_update(|_| DialogMessage::RequestComingSoonDialog { issue: Some(1091) }.into())
-					.widget_holder(),
-				IconButton::new("BooleanDifference", 24)
-					.tooltip("Coming Soon: Boolean Difference")
-					.disabled(deactivate_boolean_ops)
-					.on_update(|_| DialogMessage::RequestComingSoonDialog { issue: Some(1091) }.into())
-					.widget_holder(),
-				Separator::new(SeparatorType::Related).widget_holder(),
-				PopoverButton::new("Boolean Operations", "Coming soon").disabled(deactivate_boolean_ops).widget_holder(),
-			],
-		}]))
+		DropdownInput::new(vec![layer_selection_behavior_entries])
+			.selected_index(Some((self.tool_data.nested_selection_behavior == NestedSelectionBehavior::Shallowest) as u32))
+			.tooltip("Choose if clicking nested layers directly selects the deepest, or selects the shallowest and deepens by double clicking")
+			.widget_holder()
+	}
+
+	fn pivot_widget(&self) -> WidgetHolder {
+		PivotAssist::new(self.tool_data.pivot.to_pivot_position())
+			.on_update(|pivot_assist: &PivotAssist| SelectToolMessage::SetPivot { position: pivot_assist.position }.into())
+			.widget_holder()
+	}
+
+	fn alignment_widgets(&self) -> impl Iterator<Item = WidgetHolder> {
+		[AlignAxis::X, AlignAxis::Y]
+			.into_iter()
+			.flat_map(|axis| [(axis, AlignAggregate::Min), (axis, AlignAggregate::Center), (axis, AlignAggregate::Max)])
+			.map(|(axis, aggregate)| {
+				let (icon, tooltip) = match (axis, aggregate) {
+					(AlignAxis::X, AlignAggregate::Min) => ("AlignLeft", "Align Left"),
+					(AlignAxis::X, AlignAggregate::Center) => ("AlignHorizontalCenter", "Align Horizontal Center"),
+					(AlignAxis::X, AlignAggregate::Max) => ("AlignRight", "Align Right"),
+					(AlignAxis::Y, AlignAggregate::Min) => ("AlignTop", "Align Top"),
+					(AlignAxis::Y, AlignAggregate::Center) => ("AlignVerticalCenter", "Align Vertical Center"),
+					(AlignAxis::Y, AlignAggregate::Max) => ("AlignBottom", "Align Bottom"),
+				};
+				IconButton::new(icon, 24)
+					.tooltip(tooltip)
+					.on_update(move |_| DocumentMessage::AlignSelectedLayers { axis, aggregate }.into())
+					.widget_holder()
+			})
+	}
+
+	fn flip_widgets(&self) -> impl Iterator<Item = WidgetHolder> {
+		[(FlipAxis::X, "Horizontal"), (FlipAxis::Y, "Horizontal")].into_iter().map(|(flip_axis, name)| {
+			IconButton::new("Flip".to_string() + name, 24)
+				.tooltip("Flip ".to_string() + name)
+				.on_update(move |_| DocumentMessage::FlipSelectedLayers { flip_axis }.into())
+				.widget_holder()
+		})
+	}
```

Comment:
This is using the FlipHorizontal icon for both horizontal and vertical now.

Reply:
`(FlipAxis::Y, "Horizontal")` - typo